src/scraper.rs

```diff
@@ -182,16 +183,27 @@ impl Scraper {
 
         dom.find_urls_as_strings()
             .into_iter()
-            .filter(|candidate| Scraper::should_visit(candidate, &url))
+            .filter(|candidate| Scraper::should_visit(candidate))
             .for_each(|next_url| {
-                let mut next_full_url = url.join(&next_url).unwrap();
+                let mut url_to_parse = next_url.clone();
+                // Without this line, if url is `///tools.wmflabs.org/`, the app crashes.
+                if url_to_parse.starts_with("///") { url_to_parse = url_to_parse.replacen("///", "//", 1) }
+
+                let mut next_full_url = match url.join(url_to_parse.as_str()) {
+                    Ok(url) => url,
+                    Err(e) => panic!("FAILED TO PARSE URL: {} | Error: {}", next_url, e)
+                };
+
```

Comment:
I think this should be a separate PR

Reply:
I totally understand, but those are just a few lines. Can we please let this one slip? :)